File: Data Structures & Algorithms/validate-parentheses/submission-5.py
class Solution:
    def isValid(self, s: str) -> bool:
        matches ={'}' :'{', ')' :"(","]" :"["}
        stack = []
        if len(s) == 1:
            return False
        for ch in s:
            if ch in "{[(" :
                stack.append(ch)
            else :
                if ch in matches and stack and stack[-1] == matches[ch] :
                    stack.pop()
                else :
                    return False
            print(stack)
        if len(stack) == 0:
            return True
        else:
            return False
                    

File: Data Structures & Algorithms/minimum-stack/submission-5.js
class MinStack {
    constructor() {
        this.stack = [];
        this.minst = [];
    }

    /**
     * @param {number} val
     * @return {void}
     */
    push(val) {
        this.stack.push(val);
        if(this.minst.length == 0 || (this.minst.length > 0 && this.minst[this.minst.length-1] >= val)){
            this.minst.push(val);
        }
    }

    /**
     * @return {void}
     */
    pop() {
        if(this.stack.length >0){
            if(this.minst.length >0 && this.stack[this.stack.length-1] == this.minst[this.minst.length-1]){
                this.minst.pop();
            }
            return this.stack.pop();;
        }
    }

    /**
     * @return {number}
     */
    top() {
        if(this.stack.length >0){
            return this.stack[this.stack.length-1];
        }
    }

    /**
     * @return {number}
     */
    getMin() {
        if(this.minst.length >0){
            return this.minst[this.minst.length-1];
        }
    }
}
